### DISCUSSION TITLE
Ghostty unable to correctly render 🇹🇭 Thai text with vowels above/below the main character and tone marks, and would sometimes render overlapping characters

### DISCUSSION THREAD

**Discussion by dtinth** [ghostty-org/ghostty#3753] [Feature Requests, Ideas] [answered]
## Test case

Try to display iTerm2’s [UTF-8 demo file](https://github.com/gnachman/iTerm2/blob/master/tests/UTF-8-demo.txt) with the below command.

```sh
curl https://raw.githubusercontent.com/gnachman/iTerm2/refs/heads/master/tests/UTF-8-demo.txt
```

## Actual result on macOS Sequoia (15.1)

> [!NOTE]
> I have colored the glyphs that rendered incorrectly with red color.

### With default font

![image](https://github.com/user-attachments/assets/a8787849-ea89-4534-a528-116447bebc64)

### With `font-family = Monaco`

![image](https://github.com/user-attachments/assets/15976cc6-c030-4dc3-be96-aa9d2ff2e779)

### Terminal.app

<img width="512" alt="image" src="https://github.com/user-attachments/assets/d6d6c9d0-2c42-4526-b9fa-b43de8888bbc" />

### iTerm 2

<img width="945" alt="image" src="https://github.com/user-attachments/assets/71df5f69-fec2-4f44-b47a-b897bbc8e575" />

### VS Code terminal

<img width="661" alt="image" src="https://github.com/user-attachments/assets/db09cb83-3ec6-41e0-a3e6-6cbb97a2b188" />

**Comment by pluiedev** [comment]
Unfortunately, support for Indic scripts and other abugidas based on Indic scripts (such as the Thai script) is admittedly subpar for Ghostty as of now, since nobody in the core team speaks any of these languages or uses any of these scripts. Contributions are always welcome, though.

**Comment by pluiedev** [comment, marked as answer]
Moved to #5637
